src/wallet/wallet.h

```diff
@@ -320,6 +356,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     std::set<COutPoint> setLockedCoins;
 
     int64_t nTimeFirstKey;
+    std::map<CKeyID, CHDPubKey> mapHdPubKeys; //<! memory map of HD extended pubkeys
```

Comment:
Same here, would be better suited inside the hd chain object or an intermediary hd family object, for example:

Wallet --> HD chain ("family") --> Set of simple chains (the current HDChain object).